src/utils/useUserGeolocation.ts

```diff
@@ -0,0 +1,22 @@
+import { useState, useEffect } from 'react';
+
+const useUserGeolocation = () => {
+  const [position, setPosition] = useState<GeolocationPosition | null>(null);
+  const [error, setError] = useState<GeolocationPositionError | null>(null);
+
+  useEffect(() => {
+    const success = (position: GeolocationPosition) => {
+      setPosition(position);
+    };
+
+    const failure = (error: GeolocationPositionError) => {
+      setError(error);
+    };
+
+    navigator.geolocation.getCurrentPosition(success, failure);
+  }, []);
```

Comment:
This can be simplified:

```suggestion
    navigator.geolocation.getCurrentPosition(setPosition, setError);
  }, []);
```

Reply:
Hey thanks! That is a great suggestion!